daemon/healthz/agenthealth.go

```diff
@@ -59,47 +60,68 @@ func registerAgentHealthHTTPService(params agentHealthParams) error {
 	mux := http.NewServeMux()
 	mux.Handle("/healthz", handler)
 
-	available := len(hosts)
-	for name, host := range hosts {
-		params.JobGroup.Add(job.OneShot(fmt.Sprintf("agent-healthz-server-%s", name), func(ctx context.Context, health cell.Health) error {
```

Comment:
There's nothing wrong with using `OneShot` for long-running jobs (we've got tons of these). The issue is the `ctx` is not used in `Listen` thus we can't stop it when the job group is being stopped. Just changing `Listen(context.Background(), ...` to `Listen(ctx, ...`) should do the trick.